core/src/test/scala/unit/kafka/server/AbstractFetcherThreadTest.scala

```diff
@@ -123,4 +125,97 @@ class AbstractFetcherThreadTest {
       new DummyFetchRequest(partitionMap.map { case (k, v) => (k, v.offset) }.toMap)
   }
 
+
+  @Test
+  def testFetchRequestCorruptedMessageException() {
+    val partition = new TopicPartition("topic", 0)
+    val fetcherThread = new TestFetcherThreadForCorruptedMessage("test", "client",
+      new BrokerEndPoint(0, "localhost", 9092), 100)
```

Comment:
`MockTime` is for unit testing things which depend on the `Time` interface.
`PartitionFetchState`'s `isActive` is based on `DelayedItem`, which implements `Delayed` interface. It is unlikely we can use `MockTime` here. 
I will reduce `100` ms to `1` ms delay if you think it's OK.